src/netbase.cpp

```diff
@@ -77,15 +79,21 @@ bool static LookupIntern(const char *pszName, std::vector<CNetAddr>& vIP, unsign
     struct addrinfo aiHint;
     memset(&aiHint, 0, sizeof(struct addrinfo));
 
+    // We want a TCP port, which is a streaming socket type
     aiHint.ai_socktype = SOCK_STREAM;
     aiHint.ai_protocol = IPPROTO_TCP;
+    // We don't care which address family (IPv4 or IPv6) is returned
     aiHint.ai_family = AF_UNSPEC;
+    // If we allow lengthy lookups of hostnames, only return addresses whose
+    // family we have an address configured for.
     aiHint.ai_flags = fAllowLookup ? AI_ADDRCONFIG : AI_NUMERICHOST;
     struct addrinfo *aiRes = nullptr;
     int nErr = getaddrinfo(pszName, nullptr, &aiHint, &aiRes);
     if (nErr)
         return false;
 
+    // Traverse the linked list starting with aiTrav, add all non-internal
+    // IPv4,v6 addresses to vIP while respecting nMaxSolutions.
```

Comment:
Maybe @EthanHeilman has some experience with this?

Reply:
I would have assumed just reading this code that the OS would not impose an order on the addresses returned. However after @theuni's comment I see no reason why that assumption is justified. It seems entirely plausible that the OS or some upstream source would impose an order. Randomizing the order of the linked list would negate all of this.

Reply:
@EthanHeilman Because AF_UNSPEC actually results in two separate/parallel resolves internally, I would assume that for performance reasons the final list would simply be those two lists concatenated.

Reply:
@theuni Even if it turns out the OS randomizes the list, that make not hold in the future and/or across OSes. If we expect the lists to be randomized in this code and it seems like we do, we should randomize the lists ourselves. Seems like a neat starter project.

Reply:
@theuni @EthanHeilman My naive thought is to just pop off the i-th item in the linked list each iteration and process it, where i = random_inclusive(0, num_remaining - 1). For most cases we're only interested in 1 item, so that should be pretty cheap. We would of course have to traverse the list first to find the initial length.

Since the list fits in memory I think reservoir sampling is overkill. But it does avoid having to do linked-list manipulation, initial traversal. However, it requires us to do random number generation for each item of the list.

Is there a better way you guys are thinking of?

Reply:
I was just thinking about it with regard to seeds. Not sure what other features randomization would be important for.

Reply:
@EthanHeilman It occurred to me that this doesn't really matter until we decrease the `nMaxIPs` when learning from seeds. Right now it's 256, and no peer returns more than that, so there's no need to shuffle as there's no cut-off.